### PR TITLE
Improve User store capabilities in MI

### PR DESCRIPTION
**Pull request by milindaperera**
## Purpose
1. Add Lazy initialization for user management configs
Introduce mi.security.usermgt.lazy.init system property to enable/disable lazy init
Fix: https://github.com/wso2/micro-integrator/issues/849

2. Add JDBC user store support
Fix: https://github.com/wso2/micro-integrator/issues/865

### CODE REVIEW

File: components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/callback/AbstractPasswordCallback.java [hunk 1]
```diff
@@ -50,7 +50,12 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
             if (realmConfig == null) {
                 realmConfig = dataHolder.getRealmConfig();
                 if (realmConfig == null) {
-                    realmConfig = getRealmConfig();
+                    // If lazy loading enabled initialize security parameter
+                    if (log.isDebugEnabled()) {
+                        log.debug("Lazy loading security parameters");
+                    }
+                    ServiceComponent.initSecurityParams();
+                    realmConfig = dataHolder.getRealmConfig();
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/callback/AbstractPasswordCallback.java]: Shall we define a method for the code segment from line 52-58? I see that the code is duplicated in some other places as well

File: components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/config/RealmConfigXMLProcessor.java [hunk 2]
```diff
@@ -461,4 +461,17 @@ private OMElement getRealmElement() throws XMLStreamException, IOException, User
...
+
+
+    public static RealmConfiguration createRealmConfig() {
+        RealmConfigXMLProcessor processor = new RealmConfigXMLProcessor();
+        RealmConfiguration realmConfig = null;
+        try {
+            realmConfig = processor.buildRealmConfigurationFromFile();
+
+        } catch (org.wso2.micro.integrator.security.user.core.UserStoreException e) {
+            log.error("Error while loading Realm Configuration");
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/config/RealmConfigXMLProcessor.java]: Let's log the error aswell

File: components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/callback/AbstractPasswordCallback.java (hunk 1 above)
Comment (milindaperera) [reply components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/callback/AbstractPasswordCallback.java]: fixed, Thanks

File: components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/config/RealmConfigXMLProcessor.java (hunk 2 above)
Comment (milindaperera) [reply components/org.wso2.micro.integrator.security/src/main/java/org/wso2/micro/integrator/security/config/RealmConfigXMLProcessor.java]: Fixed, Thanks

**Pull request merged**